frame/transaction-payment/src/lib.rs

```diff
@@ -187,12 +171,39 @@ impl<T: Trait + Send + Sync> ChargeTransactionPayment<T> where
 		}
 	}
 
-	fn compute_weight_fee(weight: Weight) -> BalanceOf<T> {
+	/// Compute the fee for the specified weight.
+	///
+	/// This fee is already adjusted by the per block fee adjustment factor and is therefore
+	/// the share that the weight contributes to the overall fee of a transaction.
+	pub fn compute_weight_fee(weight: Weight) -> BalanceOf<T> where
+		BalanceOf<T>: From<u64>
+	{
+		NextFeeMultiplier::get().saturated_multiply_accumulate(
+			Self::compute_unadjusted_weight_fee(weight)
+		)
+	}
+
+	fn compute_unadjusted_weight_fee(weight: Weight) -> BalanceOf<T> {
```

Comment:
I think the naming could be better here. This function merely converts a weight to fee.

1- Either name it to reflect this and only this (not the fact that it may be unadjusted).
2- Unless if useful in other cases I'd recommend actually merging it with `compute_weight_fee `. We have one single way of converting weight to fee and that's a pipeline of a) adjustment b) weight_to_fee. easier to have it in one place.